### ISSUE TITLE
Obsolete types GraphQLHTTPResponseError et. al.

### ISSUE THREAD

**Issue by knox** [apollographql/apollo-ios#1425]
## Bug report

With the new network stack (yeah!) a few types such as `GraphQLHTTPResponseError` became obsolte but did not get removed from code base yet.

## Versions

- `apollo-ios` SDK version: 0.34.0

## Steps to reproduce

"Find usages" of `GraphQLHTTPResponseError`, `GraphQLHTTPRequestError` and probably others.

**Comment by designatednerd** [comment]
`GraphQLHTTPRequestError` is still used, but good catch on the response error. 

Please note that "find usages" isn't the be-all, end-all, especially for enum types - always worth doing a text search through the full codebase.

**Comment by knox** [comment]
The problem here for me was that by not beeing broken because of a removed class my validations got semantically broken because they where still checking for that type.

Of course you're right about `GraphQLHTTPRequestError`. I've mistaken that.

But still, regarding your advise to check for "usages" with full text search i have to strongly disagree. In fact i heavily rely on this kind of type based analysis and navigating through code and therefore i prefer AppCode over Xcode a lot because of it outstanding capabilities in that. You should give it a try.

**Comment by designatednerd** [comment]
I agree that it's helpful, but particularly when it comes to removing stuff, I've found it better to belt-and-suspenders things (even in AppCode, which I have used before).

**Comment by Max-Job** [comment]
how I can handle server error?

**Comment by knox** [comment]
Try something like this
```
        if case .invalidResponseCode(let response, _) = error as? ResponseCodeInterceptor.ResponseCodeError,
              let httpResponse = response {
            print(httpResponse.statusCode)
        }
```

**Comment by Max-Job** [comment]
@knox oh! thank you so much! it works for me)
but now I have another problem:
if environment (API URL) doesn't exist [it's](https://github.com/apollographql/apollo-ios/issues/1425#issuecomment-734286610) not working

**Comment by Max-Job** [comment]
if I write incorrect url here, I cannot handle error and get status code
```
let transport = RequestChainNetworkTransport(interceptorProvider: provider,
                                                     endpointURL: url)
        return ApolloClient(networkTransport: transport, store: store)
```

**Comment by designatednerd** [comment]
@Max-Job Could you please open this as a different issue and provide more details like: 
- What type of error are you expecting to receive in this instance? 
- What is the type of error which you do receive?
